galsim/base.py

```diff
@@ -208,15 +208,19 @@ def __init__(self, obj):
         if isinstance(obj, GSObject):
             self.SBProfile = obj.SBProfile
             if hasattr(obj,'noise'):
-                self.noise = obj.noise.copy()
-        elif isinstance(obj, galsim._galsim.SBProfile):
+                self.noise = obj.noise
+        elif isinstance(obj, _galsim.SBProfile):
             self.SBProfile = obj
         else:
             raise TypeError("GSObject must be initialized with an SBProfile or another GSObject!")
-        # a couple of definitions for using GSObjects as duck-typed ChromaticObjects
-        self.separable = True
-        self.SED = lambda w: 1.0 # flat spectrum in photons/nanometer
-        self.wave_list = np.array([], dtype=float)
+
+    # a couple of definitions for using GSObjects as duck-typed ChromaticObjects
+    @property
```

Comment:
In this case, the main reason for doing this is that lambda expressions are not picklable.  So making it a property means it doesn't actually appear in the `__dict__`, so it doesn't need to be pickled.  Doing this with all three was just for consistency to make them all look the same, but it doesn't really matter much for the other two.

The other two benefits, which are inconsequential in this case, but have been a reason for me to use `@property` in other cases are:
1. Efficiency.  The code only creates the thing to be returned when it needs it rather than on every construction.  If the attribute is rarely needed and non-trivial to make, then this can be a good reason to use a property.  I don't think either the array or the lambda are terribly time consuming to make, so it doesn't matter much here, although maybe it's a slight efficiency gain.
2. Read-only.  I don't think we are worried about users writing other values to these attributes, but in other cases, I've used properties to help enforce the immutability of some of our objects.  You can write setters for properties, but the simple thing I did here means that these are all read-only.